cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/ValidatorTransform.java

```diff
@@ -120,14 +117,21 @@ public void initialize(TransformContext context) throws Exception {
   @VisibleForTesting
   void setUpInitialScript(TransformContext context, List<Validator> validators) throws ScriptException {
     ScriptEngineManager manager = new ScriptEngineManager();
-    ScriptEngine engine = manager.getEngineByName("JavaScript");
+    engine = manager.getEngineByName("JavaScript");
     String scriptStr = config.validationScript;
     Preconditions.checkArgument(!Strings.isNullOrEmpty(scriptStr), "Filter script must be specified.");
 
     for (Validator validator : validators) {
       engine.put(validator.getValidatorName(), validator.getValidator());
     }
-    engine.eval(scriptStr);
+
+    // this is pretty ugly, but doing this so that we can pass the 'input' json into the isValid function.
+    // that is, we want people to implement
+    // function isValid(input) { ... }
+    // rather than function isValid() { ... } and have them access a global variable in the function
```

Comment:
and have them be able to access global variables in the function [is that a correct description?]